### PR TITLE
Merge Forward from 2018.3 to master

### PR DESCRIPTION
**Pull request by Ch3LL**
Conflicts:
  - .ci/docs
  - .ci/kitchen-amazon2-py2
  - .ci/kitchen-amazon2-py3
  - .ci/kitchen-centos6-py2
  - .ci/kitchen-centos7-py2
  - .ci/kitchen-centos7-py3
  - .ci/kitchen-debian8-py2
  - .ci/kitchen-debian9-py2
  - .ci/kitchen-debian9-py3
  - .ci/kitchen-fedora29-py2
  - .ci/kitchen-fedora29-py3
  - .ci/kitchen-ubuntu1604-py2
  - .ci/kitchen-ubuntu1604-py3
  - .ci/kitchen-ubuntu1804-py2
  - .ci/kitchen-ubuntu1804-py3
  - .ci/kitchen-windows2016-py2
  - .ci/kitchen-windows2016-py3
  - .ci/lint
  - .gitignore
  - .pre-commit-config.yaml
  - doc/man/salt-api.1
  - doc/man/salt-call.1
  - doc/man/salt-cloud.1
  - doc/man/salt-cp.1
  - doc/man/salt-key.1
  - doc/man/salt-master.1
  - doc/man/salt-minion.1
  - doc/man/salt-proxy.1
  - doc/man/salt-run.1
  - doc/man/salt-ssh.1
  - doc/man/salt-syndic.1
  - doc/man/salt-unity.1
  - doc/man/salt.1
  - doc/man/salt.7
  - doc/man/spm.1
  - doc/topics/releases/2018.3.5.rst
  - noxfile.py
  - pkg/osx/req.txt
  - pkg/windows/req.txt
  - requirements/static/amzn-2018.03.in
  - requirements/static/arch.txt
  - requirements/static/centos-6.txt
  - requirements/static/centos-7.in
  - requirements/static/centos-7.txt
  - requirements/static/cloud.in
  - requirements/static/debian-8.txt
  - requirements/static/debian-9.txt
  - requirements/static/fedora-28.txt
  - requirements/static/fedora-29.txt
  - requirements/static/fedora-30.in
  - requirements/static/lint.in
  - requirements/static/opensuse-42.txt
  - requirements/static/opensuse-leap-15.txt
  - requirements/static/osx.in
  - requirements/static/py2.7/cloud.txt
  - requirements/static/py2.7/lint.txt
  - requirements/static/py2.7/raet-arch.txt
  - requirements/static/py2.7/raet-centos-6.txt
  - requirements/static/py2.7/raet-debian-8.txt
  - requirements/static/py2.7/raet-debian-9.txt
  - requirements/static/py2.7/raet-fedora-30.txt
  - requirements/static/py2.7/raet-opensuse-42.txt
  - requirements/static/py2.7/raet-opensuse-leap-15.txt
  - requirements/static/py2.7/raet-ubuntu-16.04.txt
  - requirements/static/py2.7/raet-ubuntu-18.04.txt
  - requirements/static/py2.7/zeromq-amzn-2018.03.txt
  - requirements/static/py2.7/zeromq-arch.txt
  - requirements/static/py2.7/zeromq-centos-6.txt
  - requirements/static/py2.7/zeromq-centos-7.txt
  - requirements/static/py2.7/zeromq-debian-8.txt
  - requirements/static/py2.7/zeromq-debian-9.txt
  - requirements/static/py2.7/zeromq-fedora-29.txt
  - requirements/static/py2.7/zeromq-fedora-30.txt
  - requirements/static/py2.7/zeromq-opensuse-leap-15.txt
  - requirements/static/py2.7/zeromq-osx.txt
  - requirements/static/py2.7/zeromq-ubuntu-16.04.txt
  - requirements/static/py2.7/zeromq-ubuntu-18.04.txt
  - requirements/static/py2.7/zeromq-windows.txt
  - requirements/static/py3.4/cloud.txt
  - requirements/static/py3.4/zeromq-arch.txt
  - requirements/static/py3.4/zeromq-centos-7.txt
  - requirements/static/py3.4/zeromq-debian-8.txt
  - requirements/static/py3.4/zeromq-debian-9.txt
  - requirements/static/py3.4/zeromq-fedora-29.txt
  - requirements/static/py3.4/zeromq-fedora-30.txt
  - requirements/static/py3.4/zeromq-opensuse-leap-15.txt
  - requirements/static/py3.4/zeromq-ubuntu-16.04.txt
  - requirements/static/py3.4/zeromq-ubuntu-18.04.txt
  - requirements/static/py3.5/cloud.txt
  - requirements/static/py3.5/zeromq-arch.txt
  - requirements/static/py3.5/zeromq-centos-7.txt
  - requirements/static/py3.5/zeromq-debian-8.txt
  - requirements/static/py3.5/zeromq-debian-9.txt
  - requirements/static/py3.5/zeromq-fedora-29.txt
  - requirements/static/py3.5/zeromq-fedora-30.txt
  - requirements/static/py3.5/zeromq-opensuse-leap-15.txt
  - requirements/static/py3.5/zeromq-osx.txt
  - requirements/static/py3.5/zeromq-ubuntu-16.04.txt
  - requirements/static/py3.5/zeromq-ubuntu-18.04.txt
  - requirements/static/py3.5/zeromq-windows.txt
  - requirements/static/py3.6/cloud.txt
  - requirements/static/py3.6/zeromq-arch.txt
  - requirements/static/py3.6/zeromq-centos-7.txt
  - requirements/static/py3.6/zeromq-debian-8.txt
  - requirements/static/py3.6/zeromq-debian-9.txt
  - requirements/static/py3.6/zeromq-fedora-29.txt
  - requirements/static/py3.6/zeromq-fedora-30.txt
  - requirements/static/py3.6/zeromq-opensuse-leap-15.txt
  - requirements/static/py3.6/zeromq-osx.txt
  - requirements/static/py3.6/zeromq-ubuntu-16.04.txt
  - requirements/static/py3.6/zeromq-ubuntu-18.04.txt
  - requirements/static/py3.6/zeromq-windows.txt
  - requirements/static/py3.7/cloud.txt
  - requirements/static/py3.7/zeromq-arch.txt
  - requirements/static/py3.7/zeromq-centos-7.txt
  - requirements/static/py3.7/zeromq-debian-8.txt
  - requirements/static/py3.7/zeromq-debian-9.txt
  - requirements/static/py3.7/zeromq-fedora-29.txt
  - requirements/static/py3.7/zeromq-fedora-30.txt
  - requirements/static/py3.7/zeromq-opensuse-leap-15.txt
  - requirements/static/py3.7/zeromq-osx.txt
  - requirements/static/py3.7/zeromq-ubuntu-16.04.txt
  - requirements/static/py3.7/zeromq-ubuntu-18.04.txt
  - requirements/static/py3.7/zeromq-windows.txt
  - requirements/static/ubuntu-14.04.txt
  - requirements/static/ubuntu-16.04.txt
  - requirements/static/windows.in
  - salt/client/mixins.py
  - salt/crypt.py
  - salt/daemons/flo/core.py
  - salt/daemons/flo/worker.py
  - salt/daemons/flo/zero.py
  - salt/engines/http_logstash.py
  - salt/engines/logstash_engine.py
  - salt/modules/beacons.py
  - salt/modules/chocolatey.py
  - salt/modules/cron.py
  - salt/modules/file.py
  - salt/modules/gpg.py
  - salt/modules/http.py
  - salt/modules/state.py
  - salt/modules/win_file.py
  - salt/modules/win_task.py
  - salt/states/file.py
  - salt/states/pip_state.py
  - salt/states/saltmod.py
  - salt/transport/ipc.py
  - salt/transport/tcp.py
  - salt/transport/zeromq.py
  - salt/utils/cloud.py
  - salt/utils/napalm.py
  - salt/utils/schedule.py
  - salt/utils/vmware.py
  - salt/utils/yamlloader.py
  - tests/integration/__init__.py
  - tests/integration/client/test_kwarg.py
  - tests/integration/cloud/clouds/test_ec2.py
  - tests/integration/cloud/clouds/test_gogrid.py
  - tests/integration/cloud/clouds/test_oneandone.py
  - tests/integration/cloud/clouds/test_vmware.py
  - tests/integration/cloud/clouds/test_vultrpy.py
  - tests/integration/files/pillar/base/top.sls
  - tests/integration/modules/test_cp.py
  - tests/integration/modules/test_pip.py
  - tests/integration/modules/test_pkg.py
  - tests/integration/modules/test_state.py
  - tests/integration/modules/test_system.py
  - tests/integration/modules/test_win_lgpo.py
  - tests/integration/shell/test_call.py
  - tests/integration/states/test_beacon.py
  - tests/integration/states/test_pip_state.py
  - tests/integration/states/test_pkg.py
  - tests/support/generate-names-file-from-failed-test-reports.py
  - tests/support/gitfs.py
  - tests/support/helpers.py
  - tests/unit/modules/test_file.py
  - tests/unit/modules/test_gpg.py
  - tests/unit/modules/test_localemod.py
  - tests/unit/modules/test_win_file.py
  - tests/unit/modules/test_win_lgpo.py
  - tests/unit/states/test_file.py
  - tests/unit/states/test_pip_state.py
  - tests/unit/test_loader.py
  - tests/unit/utils/test_pycrypto.py
  - tests/unit/utils/test_schema.py

### CODE REVIEW

File: requirements/static/py2.7/zeromq-opensuse-42.txt [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#
```
Comment (s0undt3ch) [inline requirements/static/py2.7/zeromq-opensuse-42.txt]: All `requirements/static/py*/zeromq-*.txt` files should be deleted on master

File: salt/states/pip_state.py [hunk 2]
```diff
@@ -96,7 +96,21 @@ def pip_has_exceptions_mod(ver):
     HAS_PIP = True
 except ImportError:
     HAS_PIP = False
+    # Remove references to the loaded pip module above so reloading works
+    import sys
```
Comment (s0undt3ch) [inline salt/states/pip_state.py]: I will try and see this again tomorrow from my laptop, but I think @dwoz refactored this.

File: salt/utils/cloud.py [hunk 3]
```diff
@@ -1981,25 +1981,22 @@ def fire_event(key, msg, tag, sock_dir, args=None, transport='zeromq'):
...
+        try:
+            event.fire_event(msg, tag)
+        except ValueError:
+            # We're using at least a 0.17.x version of salt
+            if isinstance(args, dict):
+                args[key] = msg
+            else:
+                args = {key: msg}
+            event.fire_event(args, tag)
+        finally:
```
Comment (s0undt3ch) [inline salt/utils/cloud.py]: With the `with` context we no longer need this `finally` block

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/states/pip_state.py (hunk 2 above)
Comment (Ch3LL) [reply salt/states/pip_state.py]: bump @s0undt3ch did you find what you were referring to here?

Comment (s0undt3ch) [reply salt/states/pip_state.py]: This is already his code. Carry on. Sorry for the wait.

**Comment by Ch3LL** [comment]
re-run full all

**Review by waynew — APPROVED** [review]
Looking good. Wow that's a lot of merge forward 😬

**Review by garethgreenaway — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**